From 0b0c18b45d1a39f9f33172d3c22f5b8031fe564e Mon Sep 17 00:00:00 2001
From: jinweic chen <jinweic@codeaurora.org>
Date: Tue, 13 Mar 2018 13:28:30 +0800
Subject: qcacld-2.0: Fix buffer overflow in ol_rx_in_order_indication_handler

Propagation from cld3.0 to cld2.0.
Currently variable "tid" is from message, which is used directly
as array size which causes buffer over-write.

To address this issue, add check for the array size.

Change-Id: Idb6bd8ceaa217620a60bc04f2e84a551113e6edb
CRs-Fixed: 2204463
---
 CORE/CLD_TXRX/TXRX/ol_rx.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CORE/CLD_TXRX/TXRX/ol_rx.c b/CORE/CLD_TXRX/TXRX/ol_rx.c
index c5034ae..b2a6ce1 100644
--- a/CORE/CLD_TXRX/TXRX/ol_rx.c
+++ b/CORE/CLD_TXRX/TXRX/ol_rx.c
@@ -1271,6 +1271,13 @@ ol_rx_in_order_indication_handler(
     int status;
     adf_nbuf_t head_msdu, tail_msdu = NULL;
 
+    if (tid >= OL_TXRX_NUM_EXT_TIDS) {
+        TXRX_PRINT(TXRX_PRINT_LEVEL_ERR,
+                   "%s: invalid tid, %u\n", __FUNCTION__, tid);
+        WARN_ON(1);
+        return;
+    }
+
     if (pdev) {
         peer = ol_txrx_peer_find_by_id(pdev, peer_id);
         if (VOS_MONITOR_MODE == vos_get_conparam())
-- 
cgit v1.1

